File: Data Structures & Algorithms/house-robber/submission-2.cpp
class Solution {
public:
    int rob(vector<int>& nums) {
        vector<int> sol(nums.size(),0);
        for(int i=nums.size()-1;i>=0;i--){
            int maxVal=0;
            for(int k=i+2;k<nums.size();k++){
                maxVal=max(maxVal,sol[k]);
            }
            //cout<<maxVal<<" "<<nums[i]<<endl;
            sol[i]=nums[i]+maxVal;
            //cout<<sol[i]<<endl;
        }
        if(nums.size()>1)
        return max(sol[0],sol[1]);
        else return sol[0];
    }
};

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-2.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        n = len(nums)
        l, r = 0, n-1

        while l < r:
            m = (l+r)//2

            if nums[l] > nums[r] and nums[m] >= nums[l]:
                l = m + 1
            elif nums[l] > nums[r] and nums[m] < nums[l]:
                r = m
            elif nums[l] < nums[r]:
                r = l

        return nums[l]
        